### PR TITLE
#1153 export to excel

### PR DESCRIPTION
**Pull request by sarmbruster**
This PR addes couple of procedure for exporting to an Excel file using the POI library.

Note: docs are still missing. Working on this.

### CODE REVIEW

File: src/main/java/apoc/export/xls/ExportXls.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    private Stream<ProgressInfo> exportXls(@Name("file") String fileName, String source, Object data, Map<String,Object> configMap) throws Exception {
+        checkWriteAllowed();
+        try (Transaction tx = db.beginTx()) {
+            XlsExportConfig config = new XlsExportConfig(configMap);
+            ProgressInfo progressInfo = new ProgressInfo(fileName, source, "xls");
+            progressInfo.batchSize = config.getBatchSize();
+            ProgressReporter reporter = new ProgressReporter(null, null, progressInfo);
+
+            SXSSFWorkbook wb = new SXSSFWorkbook(100);
+            OutputStream out = getOutputStream(fileName, null);
```
Comment (jexp) [inline src/main/java/apoc/export/xls/ExportXls.java]: resource handling?

File: src/main/java/apoc/util/FileUtils.java [hunk 2]
```diff
@@ -124,20 +124,23 @@ public static boolean isFile(String fileName) {
     }
 
     public static PrintWriter getPrintWriter(String fileName, Writer out) throws IOException {
```
Comment (jexp) [inline src/main/java/apoc/util/FileUtils.java]: what's the reason behind these changes?

File: src/main/java/apoc/util/FileUtils.java [hunk 3]
```diff
@@ -124,20 +124,23 @@ public static boolean isFile(String fileName) {
...
+        OutputStream outputStream;
         if (isHdfs(fileName)) {
             try {
-                writer = new OutputStreamWriter(HDFSUtils.writeFile(fileName));
+                outputStream = HDFSUtils.writeFile(fileName);
             } catch (Exception e) {
                 throw new RuntimeException(e);
             }
         } else {
-            writer = fileName.equals("-") ? out : new BufferedWriter(new FileWriter(fileName));
```
Comment (jexp) [inline src/main/java/apoc/util/FileUtils.java]: we shouldn't remove the buffering

File: src/main/java/apoc/util/FileUtils.java (hunk 2 above)
Comment (sarmbruster) [reply src/main/java/apoc/util/FileUtils.java]: I couldn't get excel export workin correctly when acquiring a writer and converting it to a outputstream. With an output stream it works, so I changed this.

File: src/main/java/apoc/export/xls/ExportXls.java (hunk 1 above)
Comment (sarmbruster) [reply src/main/java/apoc/export/xls/ExportXls.java]: good point, will amend

**Comment by jexp** [comment]
Looks good thanks a lot.

To be done in a subsequent PR:

- documentation
- handling of complex(nested) types via export.json converter
